R/pkg/R/DataFrame.R

```diff
@@ -570,10 +570,17 @@ setMethod("unpersist",
 
 #' Repartition
 #'
-#' Return a new SparkDataFrame that has exactly numPartitions partitions.
-#'
+#' The following options for repartition are possible:
+#' \itemize{
+#'  \item{"Option 1"} {Return a new SparkDataFrame partitioned by
+#'                      the given columns into `numPartitions`.}
+#'  \item{"Option 2"} {Return a new SparkDataFrame that has exactly `numPartitions`.}
+#'  \item{"Option 3"} {Return a new SparkDataFrame partitioned by the given columns,
+#'                      preserving the existing number of partitions.}
```

Comment:
We support:

repartition(N)
repartition(N, col1, col2)
repartition(col1, col2)

For the third case, the number of partition is spark.sql.shuffle.partitions, not preserving the existing number of partitions.

Have I misunderstood something?